lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -84,8 +84,16 @@ def schedule
       end
 
       # reloads the schedule from redis
-      def reload_schedule!
+      def reload_schedule!(verify_with: nil, max_retries: 7)
+        retries = 0
+
         @schedule = all_schedules
+        while verify_with && retries < max_retries &&
+              !(@schedule && @schedule.eql?(verify_with))
```

Comment:
do we really care about equality here? isn't it enough to check that some schedule exists?